src/d3a/models/area/__init__.py

```diff
@@ -127,27 +128,39 @@ def area_reconfigure_event(self, **kwargs):
                 if key_in_dict_and_not_none(kwargs, 'grid_fee_constant') else 0
             grid_fee_percentage = kwargs["grid_fee_percentage"] \
                 if key_in_dict_and_not_none(kwargs, 'grid_fee_percentage') else 0
-
-            validate_area(grid_fee_percentage=grid_fee_percentage,
-                          grid_fee_constant=grid_fee_constant)
-            self._set_grid_fees(grid_fee_constant, grid_fee_percentage)
+            try:
```

Comment:
I think you can try/except once for the whole method. That way the code will become more DRY

Reply:
ok!